mantidproject/mslice
# Bin step size limits and using SofQW3 to Q conversion

Implements using `SofQWNormalisedPolygons` (`SofQW3`) to rebin data (usually from MARI or with a "rings" mapping) where pixel sizes are large before conversion to MDWorkspace with the `CopyToMD` method. For large datasets (and for indirect geometry data), the direct conversion to `|Q|` using `ConvertToMD` is still used because the centre-point rebinning of `BinMD` is not a problem in those cases.

Also, implemented a calculation of the `|Q|` limits and steps from the detector angles in the datafile at loading, which is attached to the `WorkspaceProvider` as a property. This required changing the `MainWindow` code to ensure that all widgets and presenters run the same instance of `WorkspaceProvider`. 

**To test:**

Load data (single or multiple files). If the data is direct inelastic from MARI, check that when "calculate projection" is run that `SofQWNormalisedPolygons` is called. If the data is direct inelastic with "one2one" mapping from Merlin or LET, check that `SofQWNormalisedPolygons` is *not* called - and same for Iris or Osiris data.

When plotting a slice, put in a step size of `0` or a very small number (e.g. `1e-6`) and check that it resets it back to the `optimum` step size, and displays an error message in the status bar.

Fixes #118, #28, #48, #114
